src/app/qgsrelationmanagerdialog.cpp

```diff
@@ -138,7 +138,9 @@ void QgsRelationManagerDialog::mBtnAddRelation_clicked()
       ++suffix;
     }
     relation.setId( relationId );
-    relation.addFieldPair( addDlg.references().at( 0 ).first, addDlg.references().at( 0 ).second );
+    const auto references = addDlg.references();
+    for ( int p = 0; p < references.count(); p++ )
```

Comment:
Could this be a range based for loop?

```suggestion
    for ( const auto &ref : references )
```

Reply:
obviously, done.